src/ContentValidationListener.php

```diff
@@ -213,7 +215,35 @@ public function onRoute(MvcEvent $e)
         }
 
         $inputFilter->setData($data);
+
         if ($inputFilter->isValid()) {
+            if (isset($this->config[$controllerService]['use_raw_data'])
+                    && $this->config[$controllerService]['use_raw_data'] === true) {
+                $dataContainer->setBodyParams($data);
+
+                return;
+            }
+
+            if (!$inputFilter instanceof BaseInputFilter) {
+                return;
```

Comment:
Also, we should test against the `UnknownInputsInterface`, and not `BaseInputFilter`, as the interface defines the method you need.